src/actors/DiscussionActorBase.ts

```diff
@@ -27,9 +27,7 @@ export abstract class DiscussionActorBase<
 	Data extends CommentData | DiscussionData,
 > extends EntityActorBase<Data> {
 	async listComments() {
-		// TODO: Retrieve all comments, not just the first page
-		// https://github.com/JoshuaKGoldberg/OctoGuide/issues/34
-		const response = await this.octokit.request(
+		const iterator = this.octokit.paginate.iterator(
```

Comment:
[Performance] If there are very many comments on the entity, this will retrieve them _all_. Even if the only comment we care about is very early on in that list. That's not great for performance & rate-limited consumption.

Request: can you have these iterators end pagination once they find a matching comment? 

Apologies for not mentioning this on the issue. I'd had adding a filter parameter to `listComments()` back-of-mind for myself and didn't expect someone to send a PR so quickly! 😅

Reply:
I tried to fix it. But will need to further test this.